### ISSUE TITLE
`no-ref-as-operand` rule does not work in some cases

### ISSUE THREAD

**Issue by ckvv** [vuejs/eslint-plugin-vue#1954]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:*8.23.0* 
- **eslint-plugin-vue version:*9.4.0* 
- **Node version:*v16.16.0* 
- **Operating System:*macos* 

**Please show your full configuration:**

```js
/* eslint-env node */
require("@rushstack/eslint-patch/modern-module-resolution");
module.exports = {
  root: true,
  env: {
    node: true,
  },
  globals: {
    $t: true,
  },
  extends: [
    "plugin:vue/essential",
    "eslint:recommended",
    "@vue/eslint-config-prettier",
  ],
  rules: {
    "no-console": 0,
    "no-debugger": process.env.NODE_ENV === "production" ? "warn" : "off",
    "no-async-promise-executor": "off",
    "vue/no-mutating-props": 0,
    "vue/multi-word-component-names": 0,
  },
  ignorePatterns: ["*/assets/*", "components.d.ts"],
};
```

**What did you do?**	

```js
import { ref } from 'vue';

const refObj = ref({});

const key = 'age';
refObj[key] = 18;

Object.keys(refObj);
```

**What did you expect to happen?**

**What actually happened?**

**Repository to reproduce this issue**
